### PR TITLE
Fabrica Improvements 12

### PR DESCRIPTION
**Pull request by codycooperross**
## Purpose

Adds help text for the Language fields in the Fabrica form describing the ISO 639-1 source of the selections. 

closes: datacite/datacite#1590

## Approach

Currently, lupo and the metadata schema XSD expect ISO 639-1 values, which the language selectors in the form are already built to transmit. Thus, the selectors were not changed to allow for user-inputted values. Only the help text was added to give context to the existing ISO 639-1 selections. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

**Review by svogt0511 — APPROVED** [review]
Looks good to me.  Let me know what Kristian says.

**Review by kjgarza — CHANGES_REQUESTED** [review]
I don't think this accomplished feature 12.  Because it just describes the current behavior which features 12 intends to change.

Furthermore,

> metadata schema XSD expects ISO 639-1 values

I think this comment is not accurate. Neither schema nor the XSD has any restrictions on the value of languages; it has only recommendations. @codycooperross Look at the example below, you can include any value you want under language and create a valid DOI; that passes XSD validation. 

https://bracco-bxy5xhef8-datacite.vercel.app/dois/10.24413%2Ffm67-6y98

Just a note of clarification, Feature 12 is a fabric change, but Fabrica uses multiple repositories to work; it is expected that the feature change might affect more than one repo (e.g. lupo, bolognese.)

**Comment by codycooperross** [comment]
## Purpose

Adds ability for user to submit custom language codes using the Fabrica form. 

## Approach

The DataCite Metadata Schema XSD uses a `xs:language` type for the language property, which restricts the possible language values to a set expected format. Thus, the form cannot accept any value for language. I added a validation for this format in the `doi` model that is later used in the template as well. 

I implemented a template and component for the language property that mirrors the `doi-rights` template while maintaining most of the existing language component logic. I couldn't used the `doi` model language property in the `property` property of the template because the value of the select is computed based on the model's language value. The template validation behavior thus needed to be recreated within the template.

**Review by svogt0511 — COMMENTED** [review]
I tried this locally with the change you made to Lupo.  It works fine.  2 questions:

1. As described in feature 12 of the [Product Spec](https://docs.google.com/document/d/1Tl6ctSaSG3487IMktlgfy8IZaMYVRTNzu5VObyqXCWY/edit?usp=sharing), do we need to add restrictions to the field to restrict it to a 'two-letter or three-letter language code'?  What are the restrictions.
2. Should the same autosuggest be applied to the language field of the DoiTitle component?

**Review by svogt0511 — CHANGES_REQUESTED** [review]
I meant to have the comments addressed before approval.

**Comment by codycooperross** [comment]
> 1. As described in feature 12 of the [Product Spec](https://docs.google.com/document/d/1Tl6ctSaSG3487IMktlgfy8IZaMYVRTNzu5VObyqXCWY/edit?usp=sharing), do we need to add restrictions to the field to restrict it to a 'two-letter or three-letter language code'?  What are the restrictions.

The restriction boils down to the `xs:language` type of the language property in the XSD schema. The `xs:language` type allows for language codes that meet a regex as defined here: https://www.w3.org/TR/xmlschema11-2/#language The form currently restricts to this regex using a validation in the `doi` model, but the help text recommends two-letter or three-letter language codes.

> 2. Should the same autosuggest be applied to the language field of the DoiTitle component?

I can't find my comment in the Product Spec now, but I am almost certain that Kristian clarified that the feature only pertained to the top-level language field and not the language fields in other components, like title and description. @kjgarza can you confirm?

> Also, regarding Kristian's comment, you have taken care of lupo, but he also mentioned bolognese might need to be changed, as well. I can check that out

I took a look at bolognese as well. It seems that the language value is read straight from DataCite XML without modification and without relying on ISO codes: https://github.com/datacite/bolognese/blob/b0a7df3c9dd6a45eaf56fd0e06d304e4db9b837d/lib/bolognese/readers/datacite_reader.rb#L299 The various writers also transcribe the code directly, ex. https://github.com/datacite/bolognese/blob/b0a7df3c9dd6a45eaf56fd0e06d304e4db9b837d/lib/bolognese/writers/schema_org_writer.rb#L20 and https://github.com/datacite/bolognese/blob/b0a7df3c9dd6a45eaf56fd0e06d304e4db9b837d/lib/bolognese/writers/ris_writer.rb#L18 Thus, I don't believe any bolognese changes are needed.

**Review by svogt0511 — APPROVED** [review]
I read your comments below. Awesome! Merge it!

**Comment by svogt0511** [comment]
Looks Good to me!  Don't forget to rebase before you merge.  Thanks.

**Review by kjgarza — APPROVED** [review]

**Pull request merged**
